src/upchunk.ts

```diff
@@ -150,6 +158,95 @@ export class ChunkedStreamIterable implements AsyncIterable<Blob> {
   }
 }
 
+export class ChunkedFileIterable implements ChunkedIterable {
```

Comment:
NOTE: There's some amount of (fairly simple) redundancy between `ChunkedFileIterable` and `ChunkedStreamIterable` (i.e. almost everything that is not captured in the async generator/iterable function). We _**could**_ create a common base class, but wasn't sure if that was a significant enough value add. Open to having pushback on this, though.

Reply:
Said this in Slack, but I'm personally fine with this explicitly being a band-aid. We want to delete this code if Safari ever patches the issue, so I don't see any reason to spend more time than we have to on the fallback.